### PR TITLE
Vuepress

### PR DESCRIPTION
**Pull request by abea**
## Big TODOs:
- [x] Update the generate script to generate the sidebar nav structure as well: #232 
- [x] Replace all the gitbook `hint` blocks
- [x] Fix all the broken links (hopefully programmatically)
- [x] Add a top nav
- [x] Set up redirects. We should be able to do patterned redirects.

We'll do those in separate branches now that we have something decent to branch from.

### CODE REVIEW

File: docs/advanced-topics/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<script>
```
Comment (abea) [inline docs/advanced-topics/README.md:1]: It'd be nice to eventually get a real landing page for these. That there isn't might reflect that another structure would be better.

**Comment by localghost443** [comment]
This adds a lot of items to the navigation. It's not necessarily a bad thing, but in some cases, it feels unnecessary. For example, in our `Schema Field Types` section, each sub-item has multiple sub-items (area -> example, settings).

**Comment by abea** [comment]
Good point. I might be able to reduce the layers of depth in the sidebar.

For the eventual reboot, I'm inspired by Vuepress' own docs, which separate the top level sections further, so you only have a sidebar of stuff in that section. (e.g., [guide](https://v1.vuepress.vuejs.org/guide/getting-started.html), [theme](https://v1.vuepress.vuejs.org/theme/)). To do that well we'd probably need to do at least some rearranging. But maybe not.

**Comment by abea** [comment]
Closes #236

**Comment by abea** [comment]
Closes #193

File: docs/.vuepress/theme/components/AposFeedback.vue [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    display: block;
+
+    .feedback-button:last-child {
+      margin-top: 0;
+    }
+  }
+}
+
+.feedback-button {
+  display: inline-flex;
```
Comment (localghost443) [inline docs/.vuepress/theme/components/AposFeedback.vue:153]: We don't need these flex boys

File: docs/.vuepress/theme/components/AposFeedback.vue [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  line-height: 0.9;
+  letter-spacing: -0.03em;
+  text-decoration: none;
+  color: #fff;
+  text-align: center;
+  cursor: pointer;
+  text-decoration: none;
+  transition: all 0.25s;
+
+  > span {
```
Comment (localghost443) [inline docs/.vuepress/theme/components/AposFeedback.vue:171]: I think we can trash this span -- unless it's needed for something. It has weird styles attached to it?

File: docs/.vuepress/theme/components/AposFeedback.vue [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    display: inline-block;
+    border-style: solid;
+    border-width: 1px 0;
+    border-color: transparent;
+  }
+
+  &:hover, &:focus, &.is-hover {
+    background-color: $accentDarkColor;
+
+    span {
```
Comment (localghost443) [inline docs/.vuepress/theme/components/AposFeedback.vue:181]: I think just the regular hover state of the button itself is enough....admittedly subjective

File: docs/.vuepress/theme/components/AposPageMeta.vue [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+</script>
+
+<style lang="stylus" scoped>
+.meta {
+  display: flex;
+  justify-content: space-between;
+  flex-wrap: wrap;
+  padding: 2rem 0 1rem;
+  font-size: 90%;
```
Comment (localghost443) [inline docs/.vuepress/theme/components/AposPageMeta.vue:33]: Why 90%? I get nervous with percents in this case. I like these guys approach (https://engageinteractive.co.uk/blog/em-vs-rem-vs-px).

File: docs/.vuepress/theme/components/AposFeedback.vue [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    noTxt: {
+      type: String,
+      default: 'No'
+    }
+  }
+}
+</script>
+
+<style lang="stylus" scoped>
+@keyframes fadein {
```
Comment (localghost443) [inline docs/.vuepress/theme/components/AposFeedback.vue:90]: I don't feel like `scoped` makes a ton of sense here. That fadein is fairly utilitarion and this is actually on every page, right? Just riffing tho i don't actually care

File: docs/.vuepress/theme/components/AposFeedback.vue (hunk 3 above)
Comment (abea) [reply docs/.vuepress/theme/components/AposFeedback.vue:171]: That's mimicking the styles from the main apostrophe website.

File: docs/.vuepress/theme/components/AposPageMeta.vue (hunk 5 above)
Comment (abea) [reply docs/.vuepress/theme/components/AposPageMeta.vue:33]: I just wanted it `smaller`. But I can use a rem value instead.

File: docs/.vuepress/theme/components/AposFeedback.vue (hunk 6 above)
Comment (abea) [reply docs/.vuepress/theme/components/AposFeedback.vue:90]: It's limited to this component, though.

File: docs/.vuepress/theme/components/AposFeedback.vue (hunk 4 above)
Comment (abea) [reply docs/.vuepress/theme/components/AposFeedback.vue:181]: I'm going to hold onto this since it matches the main site -- until we redo that.

**Pull request merged**